.github/workflows/lint.yml

```diff
@@ -10,32 +10,36 @@ name: Golang Linter
 
 on:
   pull_request:
+    paths:
+      - '**/*.go'
+      - 'go.mod'
+      - 'go.sum'
   push:
     branches:
       - main
       - v[0-9]+.[0-9]+.x-celestia
+    paths:
+      - '**/*.go'
+      - 'go.mod'
+      - 'go.sum'
+
 jobs:
   golangci:
     name: golangci-lint
     runs-on: ubuntu-latest
     timeout-minutes: 8
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v4
-        with:
-          go-version: '1.22'
-      - uses: technote-space/get-diff-action@v6
+      - uses: actions/checkout@v4
+      
+      - uses: actions/setup-go@v5
         with:
-          PATTERNS: |
-            **/**.go
-            go.mod
-            go.sum
-      - uses: golangci/golangci-lint-action@v3.7.0
+          go-version-file: 'go.mod'
```

Comment:
[no change needed] 😮 clever! We can remove a few explicit Go versions from CI jobs in celestia-app with this trick.

Ref: https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file